columnflow/production/cms/seeds.py

```diff
@@ -125,53 +148,44 @@ def deterministic_event_seeds_setup(
     reader_targets: InsertableDict,
 ) -> None:
     """
-    Setup function that defines the vectorized seed creation function once and stores it in the
-    py:attr:`create_seed` attribute.
+    Setup function that defines conventions methods needed during the producer function.
     """
-    def create_seed(val: int, n_hex: int = 16) -> int:
-        return int(hashlib.sha256(bytes(str(val), "utf-8")).hexdigest()[:-(n_hex + 1):-1], base=16)
-
-    # store a vectorized version (only interface, not actually simd'ing)
-    self.create_seed = np.vectorize(create_seed, otypes=[np.uint64])
-
     # store primes in array
     self.primes = np.array(primes, dtype=np.uint64)
 
-    # helper to apply a field to an array with a silent failure that only issues a warning
-    def apply_field(ak_array: ak.Array, field: str) -> ak.Array | None:
+    # helper to apply a route to an array with a silent failure that only issues a warning
+    def apply_route(ak_array: ak.Array, route: Route) -> ak.Array | None:
         try:
-            return Route(field).apply(ak_array)
+            return route.apply(ak_array)
         except ak.errors.FieldNotFoundError:
+            if not route.has_tag("optional"):
+                raise
             logger.warning_once(
-                f"{id(self)}_{field}",
-                f"optional field '{field}' not found in events chunk for seed calculation",
+                f"{id(self)}_{route}",
+                f"optional route '{route}' not found in events chunk for seed calculation",
             )
             return None
 
-    self.apply_field = apply_field
+    self.apply_route = apply_route
 
 
 @producer(
-    uses={deterministic_event_seeds},
     produces={"Jet.deterministic_seed"},
```

Comment:
I understand the advantages of removing the `deterministic_event_seeds` producer from the requirements, but this implementation still accesses the `"deterministic_seed"` column, so we should at least add that to the `uses`.

Reply:
I think this would fail if the input file does not already contain a column `deterministic_seed`.

Now that I'm thinking about that, this check should actually be changed to consider columns that are produced along the way. We have all information necessary for detecting that, but I guess this would be another PR (we can talk about this offline).